compiler/src/pretyping.ml

```diff
@@ -479,6 +529,39 @@ end  = struct
 
   end
 
+  module TypeAlias = struct 
+
+    let push (env:'asm env) (id:A.pident) (ty:P.pty) : 'asm env =
+      match find (fun x -> x.gb_types) (L.unloc id) env with
+      | None -> 
+        begin
+          let ty = L.mk_loc (L.loc id) ty in
+          let doit v = {v with gb_types = Map.add (L.unloc id) ty v.gb_types }
+          in let binds = 
+          match env.e_bindings with 
+          | ([],gb) -> [],doit gb
+          | ((ns,gb):: stack, glob) -> (ns,doit gb):: stack , glob
+          in
+          {env with e_bindings = binds}
+        end 
+      | Some alias ->
+        rs_tyerror  ~loc:(L.loc id)  (DuplicateAlias (L.unloc id, (L.loc alias))) 
+
+
+    let get (env: 'asm env) (id:A.pident) : (P.pty L.located) = 
+      let typea = find (fun b -> b.gb_types) (L.unloc id) env in 
+      match typea with 
+      | None -> 
+        rs_tyerror  ~loc:(L.loc id) (TypeNotFound (L.unloc id))
+      | Some e -> e
+
+    let get_opt (env: 'asm env) (id:A.pident) : (P.pty L.located) option = 
+      match get env id with 
+      | exception _ -> None 
+      | e -> Some e
+
```

Comment:
This is `find`.

Reply:
I did it this way in case that the Env.Typealias.get function is modified in the future

Reply:
For example if we resolve type lazily, we would need to add another map to track unfound alias, making `get` behavior different

Reply:
This function doesn't exist anymore so this is closed by default